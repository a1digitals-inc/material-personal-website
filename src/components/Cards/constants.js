export const FRONTEND_TYPE = "FRONT-END";
export const BACKEND_TYPE = "BACK-END";
