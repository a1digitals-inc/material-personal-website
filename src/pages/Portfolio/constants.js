export const ALL_FILTER = "ALL";
export const FRONTEND_FILTER = "FRONTEND";
export const BACKEND_FILTER = "BACKEND";
