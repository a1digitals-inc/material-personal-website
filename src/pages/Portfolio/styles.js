import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
  filter: {
    flexGrow: 1,
    maxWidth: 500,
    margin: "0 auto",
  },
});

export default useStyles;
