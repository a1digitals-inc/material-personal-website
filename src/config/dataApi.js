import defaultprofilepic from "../assets/images/defaultprofilepic.jfif";
import * as portfolioTypes from "../components/Cards/constants";

export const personalData = {
  greeting: "Hi, I'm",
  name: "Dyarlen Iber",
  description:
    "I'm a Full-Stack developer with professional experience working with PHP, Laravel, JavaScript, Node.js, SQL, React and Go. Take a look at my work or get in touch!",
  email: "dyarlen1@gmail.com",
  profilePicture: defaultprofilepic,
  socialMedias: [
    {
      name: "github",
      url: "https://github.com/dyarleniber",
    },
    {
      name: "linkedin",
      url: "https://ie.linkedin.com/in/dyarleniber",
    },
    {
      name: "instagram",
      url: "https://www.instagram.com/dyarleniber/",
    },
    {
      name: "dev",
      url: "https://dev.to/dyarleniber",
    },
  ],
};

export const menuItems = [
  {
    to: "/",
    primary: "Home",
  },
  {
    to: "/portfolio",
    primary: "Portfolio",
  },
];

export const portfolioData = {
  title: "Portfolio",
  description:
    "All my projects are covered by automated tests and deployed on platforms such as GitHub Pages or Netlify. For more details about the source code, automated tests, CI/CD workflow and technologies used, please visit the github page of each project.",
};

const frontEndPortfolio = [
  {
    id: 101,
    name: "Personal website",
    description:
      "ðŸ’» Personal website using Javascript, React, Hooks, Material-UI and Jest.",
    type: portfolioTypes.FRONTEND_TYPE,
    technologies: ["Javascript", "React", "Hooks", "Material-UI", "Jest"],
    homepage: "https://www.dyarleniber.com",
    github: "https://github.com/dyarleniber/personal-website",
  },
  {
    id: 102,
    name: "New York Times movie reviews",
    description:
      "Application for search New York Times movie reviews and get movie critics using Javascript, React, Redux, React hooks, redux-saga and styled-components.",
    type: portfolioTypes.FRONTEND_TYPE,
    technologies: [
      "React",
      "Redux",
      "Hooks",
      "redux-saga",
      "styled-components",
    ],
    homepage: null,
    github: "https://github.com/dyarleniber/nyt-movie-reviews",
  },
  {
    id: 103,
    name: "Habit tracker front-end",
    description:
      "Habit tracker front-end system using Javascript, React, Redux, React hooks, redux-saga and styled-components.",
    type: portfolioTypes.FRONTEND_TYPE,
    technologies: [
      "React",
      "Redux",
      "Hooks",
      "redux-saga",
      "styled-components",
    ],
    homepage: null,
    github: "https://github.com/dyarleniber/habit-tracker-front-end",
  },
  {
    id: 104,
    name: "Smart brain",
    description: "Udemy Final Project.",
    type: portfolioTypes.FRONTEND_TYPE,
    technologies: ["React", "Redux"],
    homepage: null,
    github: "https://github.com/dyarleniber/smart-brain",
  },
  {
    id: 105,
    name: "To-do list",
    description: "Simple To-do list with React, Redux and Hooks.",
    type: portfolioTypes.FRONTEND_TYPE,
    technologies: ["React", "Redux", "Hooks", "Bootstrap"],
    homepage: null,
    github: "https://github.com/dyarleniber/react-to-do-list",
  },
];

const backEndPortfolio = [
  {
    id: 201,
    name: "Habit tracker back-end",
    description:
      "Habit tracker back-end system using Javascript, Node.js, Express, JWT, MongoDB, Mongoose, Nodemailer, Handlebars, Redis, Winston and Jest with 100% code coverage in tests.",
    type: portfolioTypes.BACKEND_TYPE,
    technologies: ["Node.js", "Express", "JWT", "MongoDB", "Redis", "Jest"],
    homepage: null,
    github: "https://github.com/dyarleniber/habit-tracker-back-end",
  },
  {
    id: 202,
    name: "Payment API",
    description: "Payment gateway API.",
    type: portfolioTypes.BACKEND_TYPE,
    technologies: ["PHP"],
    homepage: null,
    github: "https://github.com/dyarleniber/php-payment-api",
  },
  {
    id: 203,
    name: "Smart brain API",
    description: "Final project for Udemy course.",
    type: portfolioTypes.BACKEND_TYPE,
    technologies: ["Node.js", "Express", "JWT", "Redis"],
    homepage: null,
    github: "https://github.com/dyarleniber/smart-brain-api",
  },
  {
    id: 204,
    name: "PHP Docker environment",
    description: "PHP and MySQL environment with Docker and Docker Compose.",
    type: portfolioTypes.BACKEND_TYPE,
    technologies: ["PHP", "MySQL", "Docker"],
    homepage: null,
    github: "https://github.com/dyarleniber/docker-php",
  },
  {
    id: 205,
    name: "Awesome Fizzbuzz",
    description:
      "This project was developed applying the Chain of Responsibility Pattern and the Builder Pattern to develop FizzBuzz.",
    type: portfolioTypes.BACKEND_TYPE,
    technologies: ["PHP", "Docker"],
    homepage: null,
    github: "https://github.com/dyarleniber/awesome-fizzbuzz",
  },
];

export const portfolio = [...backEndPortfolio, ...frontEndPortfolio];
