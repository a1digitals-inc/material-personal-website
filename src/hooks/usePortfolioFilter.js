import { useState, useEffect } from "react";

import * as filters from "../pages/Portfolio/constants";
import * as types from "../components/Cards/constants";

export const handleFilter = (portfolioItems, filter) =>
  portfolioItems.filter((item) => {
    if (filter === filters.FRONTEND_FILTER) {
      return item.type === types.FRONTEND_TYPE;
    }
    if (filter === filters.BACKEND_FILTER) {
      return item.type === types.BACKEND_TYPE;
    }
    return true;
  });

const handleOrder = (portfolioItems) =>
  portfolioItems.sort(function (a, b) {
    if (a.id > b.id) {
      return 1;
    }
    return -1;
  });

function usePortfolioFilter(portfolioItems) {
  const [filter, setFilter] = useState(filters.ALL_FILTER);
  const [filteredPortfolio, setFilteredPortfolio] = useState([]);

  useEffect(() => {
    setFilteredPortfolio(handleFilter(portfolioItems, filter));
    setFilteredPortfolio((prevState) => handleOrder(prevState));
  }, [filter]);

  return { filteredPortfolio, filter, setFilter };
}

export default usePortfolioFilter;
