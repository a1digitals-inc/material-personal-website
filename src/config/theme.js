import { createMuiTheme, responsiveFontSizes } from "@material-ui/core/styles";
import grey from "@material-ui/core/colors/grey";

const blackColor = "#000";
const whiteColor = "#fff";

let theme = createMuiTheme({
  palette: {
    primary: {
      light: grey[300],
      main: grey[500],
      dark: grey[700],
    },

    secondary: {
      light: grey[300],
      main: grey[500],
      dark: grey[700],
    },

    text: {
      primary: blackColor,
      secondary: blackColor,
      disabled: blackColor,
    },

    action: {
      active: blackColor,
    },

    background: {
      default: grey[100],
      paper: whiteColor,
    },
  },
});

theme = responsiveFontSizes(theme);

export default theme;
