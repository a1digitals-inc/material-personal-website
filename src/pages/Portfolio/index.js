import React, { useState, useEffect } from "react";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";
import Grid from "@material-ui/core/Grid";

import { portfolio } from "../../config/dataApi";
import * as filters from "./constants";
import Card from "../../components/Cards";
import * as types from "../../components/Cards/constants";

const Portfolio = () => {
  const [filter, setFilter] = useState(filters.ALL_FILTER);
  const [filteredPortfolio, setFilteredPortfolio] = useState([]);

  const handleChange = (event, newValue) => {
    setFilter(newValue);
  };

  useEffect(() => {
    setFilteredPortfolio(
      portfolio.filter((item) => {
        if (filter === filters.FRONTEND_FILTER) {
          return item.type === types.FRONTEND_TYPE;
        }
        if (filter === filters.BACKEND_FILTER) {
          return item.type === types.BACKEND_TYPE;
        }
        return true;
      })
    );
    setFilteredPortfolio((prevState) => {
      return prevState.sort(function (a, b) {
        if (a.id > b.id) {
          return 1;
        }
        if (a.id < b.id) {
          return -1;
        }
        return 0;
      });
    });
  }, [filter]);

  return (
    <Grid container spacing={4}>
      <Grid item xs={12}>
        <Tabs
          value={filter}
          onChange={handleChange}
          variant="fullWidth"
          indicatorColor="primary"
          textColor="primary"
          aria-label="porfolio filter"
        >
          <Tab value={filters.ALL_FILTER} label="ALL" />
          <Tab value={filters.FRONTEND_FILTER} label="FRONT-END" />
          <Tab value={filters.BACKEND_FILTER} label="BACK-END" />
        </Tabs>
      </Grid>
      <Grid item xs={12}>
        <Grid container justify="center" spacing={4}>
          {filteredPortfolio.map((portfolioItem) => (
            <Grid key={portfolioItem.id} item>
              <Card {...portfolioItem} />
            </Grid>
          ))}
        </Grid>
      </Grid>
    </Grid>
  );
};

export default Portfolio;
