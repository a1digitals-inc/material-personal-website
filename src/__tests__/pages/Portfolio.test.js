import React from "react";
import { shallow } from "enzyme";

import Portfolio from "../../pages/Portfolio";
import * as filters from "../../pages/Portfolio/constants";
import usePortfolioFilter from "../../hooks/usePortfolioFilter";
import { portfolio } from "../../config/dataApi";

let wrapper;

jest.mock("../../hooks/usePortfolioFilter");

beforeEach(() => {
  jest.clearAllMocks();
});

it("renders without crashing", () => {
  const filteredPortfolio = portfolio;
  const filter = filters.ALL_FILTER;
  const setFilter = jest.fn();

  usePortfolioFilter.mockImplementation(() => {
    return {
      filteredPortfolio,
      filter,
      setFilter,
    };
  });

  wrapper = shallow(<Portfolio />);

  expect(wrapper).toMatchSnapshot();
});

it("sets filter", () => {
  const filteredPortfolio = [];
  const filter = filters.ALL_FILTER;
  const setFilter = jest.fn();

  usePortfolioFilter.mockImplementation(() => {
    return {
      filteredPortfolio,
      filter,
      setFilter,
    };
  });

  wrapper = shallow(<Portfolio />);

  wrapper
    .find("#tab-filter")
    .first()
    .props()
    .onChange(null, filters.FRONTEND_FILTER);

  expect(setFilter).toHaveBeenCalledTimes(1);
  expect(setFilter).toHaveBeenCalledWith(filters.FRONTEND_FILTER);
});
