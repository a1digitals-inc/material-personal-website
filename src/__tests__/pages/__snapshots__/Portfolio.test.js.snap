// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={4}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Tabs))
      aria-label="porfolio filter"
      id="tab-filter"
      indicatorColor="primary"
      onChange={[Function]}
      textColor="primary"
      value="ALL"
      variant="fullWidth"
    >
      <WithStyles(ForwardRef(Tab))
        label="ALL"
        value="ALL"
      />
      <WithStyles(ForwardRef(Tab))
        label="FRONT-END"
        value="FRONTEND"
      />
      <WithStyles(ForwardRef(Tab))
        label="BACK-END"
        value="BACKEND"
      />
    </WithStyles(ForwardRef(Tabs))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="center"
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="201"
      >
        <Cards
          description="Habit tracker back-end system using Javascript, Node.js, Express, JWT, MongoDB, Mongoose, Nodemailer, Handlebars, Redis, Winston and Jest with 100% code coverage in tests."
          github="https://github.com/dyarleniber/habit-tracker-back-end"
          homepage={null}
          id={201}
          name="Habit tracker back-end"
          technologies={
            Array [
              "Node.js",
              "Express",
              "JWT",
              "MongoDB",
              "Redis",
              "Jest",
            ]
          }
          type="BACK-END"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="202"
      >
        <Cards
          description="Payment gateway API."
          github="https://github.com/dyarleniber/php-payment-api"
          homepage={null}
          id={202}
          name="Payment API"
          technologies={
            Array [
              "PHP",
            ]
          }
          type="BACK-END"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="203"
      >
        <Cards
          description="Final project for Udemy course."
          github="https://github.com/dyarleniber/smart-brain-api"
          homepage={null}
          id={203}
          name="Smart brain API"
          technologies={
            Array [
              "Node.js",
              "Express",
              "JWT",
              "Redis",
            ]
          }
          type="BACK-END"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="204"
      >
        <Cards
          description="PHP and MySQL environment with Docker and Docker Compose."
          github="https://github.com/dyarleniber/docker-php"
          homepage={null}
          id={204}
          name="PHP Docker environment"
          technologies={
            Array [
              "PHP",
              "MySQL",
              "Docker",
            ]
          }
          type="BACK-END"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="205"
      >
        <Cards
          description="This project was developed applying the Chain of Responsibility Pattern and the Builder Pattern to develop FizzBuzz."
          github="https://github.com/dyarleniber/awesome-fizzbuzz"
          homepage={null}
          id={205}
          name="Awesome Fizzbuzz"
          technologies={
            Array [
              "PHP",
              "Docker",
            ]
          }
          type="BACK-END"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="101"
      >
        <Cards
          description="Application for search New York Times movie reviews and get movie critics using Javascript, React, Redux, React hooks, redux-saga and styled-components."
          github="https://github.com/dyarleniber/nyt-movie-reviews"
          homepage={null}
          id={101}
          name="New York Times movie reviews"
          technologies={
            Array [
              "React",
              "Redux",
              "Hooks",
              "redux-saga",
              "styled-components",
            ]
          }
          type="FRONT-END"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="102"
      >
        <Cards
          description="Habit tracker front-end system using Javascript, React, Redux, React hooks, redux-saga and styled-components."
          github="https://github.com/dyarleniber/habit-tracker-front-end"
          homepage={null}
          id={102}
          name="Habit tracker front-end"
          technologies={
            Array [
              "React",
              "Redux",
              "Hooks",
              "redux-saga",
              "styled-components",
            ]
          }
          type="FRONT-END"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="103"
      >
        <Cards
          description="Udemy Final Project."
          github="https://github.com/dyarleniber/smart-brain"
          homepage={null}
          id={103}
          name="Smart brain"
          technologies={
            Array [
              "React",
              "Redux",
            ]
          }
          type="FRONT-END"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="104"
      >
        <Cards
          description="Simple To-do list with React, Redux and Hooks."
          github="https://github.com/dyarleniber/react-to-do-list"
          homepage={null}
          id={104}
          name="To-do list"
          technologies={
            Array [
              "React",
              "Redux",
              "Hooks",
              "Bootstrap",
            ]
          }
          type="FRONT-END"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="105"
      >
        <Cards
          description="My personal website."
          github="https://github.com/dyarleniber/personal-website"
          homepage="https://dyarleniber.github.io/personal-website/"
          id={105}
          name="Personal website"
          technologies={
            Array [
              "React",
              "Hooks",
              "Material-UI",
            ]
          }
          type="FRONT-END"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
